push-payload/index.js

```diff
@@ -1,69 +1,84 @@
-var endpoint;
-var key;
-var authSecret;
+// This function is needed because Chrome doesn't accept a base64 encoded string
+// as value for applicationServerKey in pushManager.subscribe yet
+// https://bugs.chromium.org/p/chromium/issues/detail?id=802280
+function urlBase64ToUint8Array(base64String) {
```

Comment:
Maybe we can put this function in a utils.js script in the top-directory, which we include in each recipe before index.js. What do you say?

Reply:
That seems like the cleanest solution.. Unfortunately I'm not sure how to do this. If we put the file in the top-directory, it isn't served by the server process. I think it has to be specified in `gulpfile.js` to be served? Also not ideal.. Is there another way?

Reply:
Yes, probably the `gulpfile.js` has to copy the file to `dist/`.
@delapuente any other idea how to do this?

Reply:
1620dfc works, but might not be the best solution.

Reply:
That looks sane enough, maybe move it to the top repo directory though instead of src/js/.
I'm still undecided whether the best solution is to have the function in a separate file or not. One one hand, it's a pain to copy and paste it in every recipe, on the other hand, the recipes are supposed to be self-contained.

Reply:
(Sorry if the discussion is taking long, but once we have decided here we will apply the same rules to all recipes, so it will be way faster then).

Reply:
Yep I agree that it's both not optimal.. Its a shame that Chrome hasn't even started their native implementation yet.. otherwise we could maybe leave the function out completely. But since it's not even on the horizon that's not really a possibility either.

(I've moved tools.js to the top directory in 1bb1db5)